charts/nextcloud/templates/_helpers.tpl

```diff
@@ -156,6 +156,10 @@ Create environment variables used to configure the nextcloud container as well a
 - name: NEXTCLOUD_UPDATE
   value: {{ .Values.nextcloud.update | quote }}
 {{- end }}
+{{- if .Values.nextcloud.initLock }}
```

Comment:
Shouldn't we just enable it if replicas > 0? I don't see the point in having an option for it, since it will do nothing in a single replica environment and is required for multiple replica environments.

Reply:
The number of replicas can be changed at any time though, using `kubectl scale`, or by horizontal pod autoscaler. If the chart makes a decision when you install/upgrade, that decision can become wrong.

Reply:
I see, then let's add a big warning that the lock should be enabled when having multiple replicas. We could also just always enable it, since it doesn't hurt setups with only a single replica.

Reply:
I agree with you there, I just used the same default as the Docker image. There is the possibility of a lockup if the upgrade fails (because the lock file is left behind), I think that's why it's disabled by default, but I have an open PR against the Docker image to avoid that: https://github.com/nextcloud/docker/issues/1903